src/graph.py

```diff
@@ -12,32 +16,41 @@ class NoOverlapError(PathOverlapError):
 class NodeMissingError(ValueError):
     pass
 
+
 class Node:
-    def __init__(self, seq: str, paths: Iterable[int]):
+    def __init__(self, seq: str, paths: 'Iterable[Path]', id: str = None):
         assert isinstance(seq, str), seq
-        assert not isinstance(paths, str) and isinstance(paths, Iterable), paths
+        self.id = id if id else str(uuid1())
         self.seq = seq
-        self.paths = set(paths)
+        self.paths = set()  # Set[PathIndex]
+        for p in paths:
+            self.append_path(p)
 
     def __len__(self):
         return len(self.paths)
 
     def __repr__(self):
         """Paths representation is sorted because set ordering is not guaranteed."""
         return repr(self.seq) + \
-        ', {' + ', '.join(str(i) for i in sorted(list(self.paths))) + '}'
+        ', {' + ', '.join(str(i) for i in list(self.paths)) + '}'
 
     def __eq__(self, other):
         if not isinstance(other, Node):
             print("Warn: comparing Node and ", type(other), other)
             return False
-        return self.seq == other.seq and self.paths == other.paths
+        return self.seq == other.seq and self.paths == other.paths  # and self.id == other.id
```

Comment:
It might be better to clarify the intention of method. Considering the case that there are duplicated nodes on multiple paths, does this method intend to compare whether the node is identically the same object on GFA, or the equality of the  sequence?
If you do not hope to check the node equalities by node id, then comparing the position on the path (i.e. rank).

Reply:
You're right to question this.  The issue is when uuid1() is used to generate when an id is not provided, they won't match even though the topology is the same.  I decided the topology was more important than how it was generated.  You could add this as a comment.